api/v1alpha1/ext_auth_types.go

```diff
@@ -78,4 +78,11 @@ type HTTPExtAuthService struct {
 	// original client request.
 	// +optional
 	HeadersToBackend []string `json:"headersToBackend,omitempty"`
+
+	// HeadersToMetadata are the authorization response headers that will be emitted
+	// as dynamic metadata to be consumed by next filter.
+	// Note: this metadata lives in envoy.filters.http.ext_authz and can only be consumed
+	// by the next filter if explicitly specified.
+	// +optional
+	HeadersToMetadata []string `json:"headersToMetadata,omitempty"`
```

Comment:
looks like only support exact with this PR, what will happen if we want to support prefix/suffix/regex in the future?

BTW, please proposal the API first.

Reply:
yes, this will only support exact matching for now, similar to `HeadersToBackend`.

Apologies, I wasn't aware a proposal was needed. I have now created it on https://github.com/envoyproxy/gateway/issues/4163.

Reply:
we will need to change the API for `HeadersToBackend` also if we want to support prefix/suffix/regex in the future, do you think we need it @zirain? if so I'll try to change the implementation for both

Reply:
let's focus on metadata first, WDYT?

Reply:
I believe it may be sufficient to just support the exact match for the `HeadersToMetadata`, considering that
* HTTP header names are typically fixed and don't change dynamically, and the number of HTTP headers is usually quite limited
* A downstream filter that needs to extract the value from a Metada will also need to know the Metadata name in advance, so having a fixed name seems reasonable

Supporting prefix/suffix/regex may unnecessarily complicate the API.

That said, I could be mistaken on this.  @envoyproxy/gateway-maintainers, please feel free to chime in.

Reply:
yes, I think adding prefix, suffix, or regex might complicate the API. For now, could we consider only supporting exact matches @zirain?